pkg/daemon/criruntime/containerruntime/docker.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package containerruntime
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	dockerapi "github.com/docker/docker/client"
+	daemonutil "github.com/openkruise/kruise/pkg/daemon/util"
+	"github.com/openkruise/kruise/pkg/util"
+)
+
+func NewDockerContainerService(runtimeURI string) (ContainerService, error) {
+	d := &dockerContainerService{runtimeURI: runtimeURI}
+	if err := d.createRuntimeClientIfNecessary(); err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+type dockerContainerService struct {
+	sync.Mutex
+	runtimeURI string
+	client     *dockerapi.Client
+}
+
+func (d *dockerContainerService) createRuntimeClientIfNecessary() error {
+	d.Lock()
+	defer d.Unlock()
+	if d.client != nil {
+		return nil
+	}
+	c, err := dockerapi.NewClient(d.runtimeURI, "1.23", nil, nil)
```

Comment:
`1.23`, the runtime version is hard coded, consider passing it via the `runtimeConfig` or another better way.

Reply:
Thanks. Actually, this version indicates the version of the client (kruise-daemon), not the server (the docker-daemon running on node). So it is more related to how we use the client to get containers and images, and the server API should be compatible.

Reply:
Oh, I got it, thanks very much!